Tokens not aligned error when spacy < 2.3.0

## Description of the bug

"Tokens are not aligned!" RuntimeError happens at `HocrVisualParser.parse`

## To Reproduce
Steps to reproduce the behavior:
1. Use the latest commit (a2d097be6f6509214a93246ad8027aeed509ba65)
2. Parse [William_R._King.hocr.txt](https://github.com/HazyResearch/fonduer/files/5427085/William_R._King.hocr.txt) with `HOCRDocPreprocessor` and `HocrVisualParser`

## Expected behavior

Parse without an error

## Error Logs/Screenshots

```
[ERROR] Document William_R._King not added to database, because of parse error: 
('Tokens are not aligned!', 'hocr tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon.\', \'William\', \'R.\', \'King,\', \'of\', \'Alabama,\', \'vice-president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States,\', \'eighth\', \'of\', \'December,\', "1853\'\'"]', 'spacy tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King\', \'?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon\', \'.\', \'William\', \'R.\', \'King\', \',\', \'of\', \'Alabama\', \',\', \'vice\', \'-\', \'president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States\', \',\', \'eighth\', \'of\', \'December\', \',\', \'1853\', "\'", "\'"]')
Traceback (most recent call last):
  File "/Users/hiromu/workspace/fonduer/src/fonduer/parser/parser.py", line 246, in apply
    document.name, document.sentences
  File "/Users/hiromu/workspace/fonduer/src/fonduer/parser/parser.py", line 244, in <listcomp>
    y
  File "/Users/hiromu/workspace/fonduer/src/fonduer/parser/visual_parser/hocr_visual_parser.py", line 134, in parse
    f"spacy tokens: {spacy_tokens}",
RuntimeError: ('Tokens are not aligned!', 'hocr tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon.\', \'William\', \'R.\', \'King,\', \'of\', \'Alabama,\', \'vice-president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States,\', \'eighth\', \'of\', \'December,\', "1853\'\'"]', 'spacy tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King\', \'?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon\', \'.\', \'William\', \'R.\', \'King\', \',\', \'of\', \'Alabama\', \',\', \'vice\', \'-\', \'president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States\', \',\', \'eighth\', \'of\', \'December\', \',\', \'1853\', "\'", "\'"]')
```

## Environment (please complete the following information)
 - OS: macOS 10.15.6
 - PostgreSQL Version: N/A
 - Poppler Utils Version: N/A
 - Fonduer Version: 0.8.4+dev (a2d097be6f6509214a93246ad8027aeed509ba65)
 - spaCy: 2.2.4

## Additional context
Add any other context about the problem here.

This error does not happen when spacy >= 2.3.0

--- a/src/fonduer/parser/visual_parser/hocr_visual_parser.py
+++ b/src/fonduer/parser/visual_parser/hocr_visual_parser.py
@@ -23,14 +23,13 @@ class HocrVisualParser(VisualParser):
     ):
         """Initialize a visual parser.
 
-        :raises ImportError: an error is raised when spaCy is not 2.2.2 or later.
+        :raises ImportError: an error is raised when spaCy is not 2.3.0 or later.
         """
-        if version.parse(spacy.__version__) < version.parse("2.2.2"):
+        if version.parse(spacy.__version__) < version.parse("2.3.0"):
             raise ImportError(
-                f"You are using spaCy {spacy.__version__},"
-                f"but it should be 2.2.2 or later."
+                f"You are using spaCy {spacy.__version__}, "
+                f"but it should be 2.3.0 or later to use HocrVisualParser."
             )
-        spacy.gold.USE_NEW_ALIGN = True
         self.replacements: List[Tuple[Pattern, str]] = []
         for (pattern, replace) in replacements:
             self.replacements.append((re.compile(pattern, flags=re.UNICODE), replace))

--- a/tests/data/hocr_simple/121.hocr
+++ b/tests/data/hocr_simple/121.hocr
@@ -51,7 +51,8 @@
           <span class="ocrx_word" id="word_1_67" title="bbox 358 193 366 201">of</span>
           <span class="ocrx_word" id="word_1_68" title="bbox 369 193 384 201">this</span>
           <span class="ocrx_word" id="word_1_69" title="bbox 387 193 426 201">Account.</span>
-          <span class="ocrx_word" id="word_1_70" title="bbox 430 193 453 202">“We,”</span>
+          <!--Use two single-quotes to see if #527 happens-->
+          <span class="ocrx_word" id="word_1_70" title="bbox 430 193 453 202">''We,''</span>
           <span class="ocrx_word" id="word_1_71" title="bbox 457 193 478 202">“Us,”</span>
           <span class="ocrx_word" id="word_1_72" title="bbox 482 193 504 201">“Our”</span>
           <span class="ocrx_word" id="word_1_73" title="bbox 508 193 524 201">and</span>

